src/control/server/server_utils.go

```diff
@@ -38,6 +38,8 @@ type netListenFn func(string, string) (net.Listener, error)
 // resolveTCPFn is a type alias for the net.ResolveTCPAddr function signature.
 type resolveTCPFn func(string, string) (*net.TCPAddr, error)
 
+var hostnameCached string
```

Comment:
It does seem a little strange to be adding a new global variable for this. Why not query the hostname once at startup and then stash it in the server struct?